### PR TITLE
Add possibility to choose config file, implement temporary sessions

### PR DESCRIPTION
**Pull request by xates**
- Added the possibility to choose a configuration file by providing its path when initializing the `Client`, instead of using the previously hard-coded "config.ini" file
- Allowing to set the `session_name` to `None` in order to make it "temporary", so that it won't be saved

### CODE REVIEW

**Comment by delivrance** [comment]
Hi. I wonder how temporary sessions can be useful, you have to log-in with phone number and code every single time if you don't save them.

File: pyrogram/client/client.py [hunk 1]
```diff
@@ -74,7 +78,8 @@ class Client:
     INVITE_LINK_RE = re.compile(r"^(?:https?:\/\/)?t\.me\/joinchat\/(.+)$")
     DIALOGS_AT_ONCE = 100
 
-    def __init__(self, session_name: str, test_mode: bool = False):
+    def __init__(self, config_file: str = "config.ini", session_name: str = None, test_mode: bool = False):
```
Comment (delivrance) [inline pyrogram/client/client.py]: Maybe we should keep `session_name` as the first parameter so you don't have to explicitly pass a kwarg every time.

**Comment by xates** [comment]
Hi! For example, when the user is logging in on a public/shared device and it is just a one-time login and he doesn't want to save any data.

File: pyrogram/client/client.py (hunk 1 above)
Comment (xates) [reply pyrogram/client/client.py]: That is ok for me. As you like it best.

Comment (xates) [reply pyrogram/client/client.py]: It just seemed me conceptually right to have the config file first :smile:

**Comment by LeonardoRick** [comment]
Could you guys give an example on how to use this config.ini or another solution to change `.session` files location? Thanks a lot!

**Pull request merged**